connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -305,7 +305,11 @@ class SparkConnectPlanner(session: SparkSession) {
   }
 
   private def transformHint(rel: proto.Hint): LogicalPlan = {
-    val params = rel.getParametersList.asScala.map(toCatalystValue).toSeq
+    val params = rel.getParametersList.asScala.map(toCatalystValue).toSeq.map{
+      case name: String => UnresolvedAttribute(name)
+      case l: Long => l.toInt
```

Comment:
I am a bot confused here: why need this convert?

Reply:
Python only has int type and has no long type, so we can't pass in a int literal since we always assumes the largest type.

Reply:
sorry if I am being dumb but why do we need pass a int literal?

Reply:
Hi @HyukjinKwon , using a long value will fail scala pattern match here since it will only accept a int value:

https://github.com/apache/spark/blob/c78f935185cbfae2e45adb393def0bd389a89c7e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala#L242-L249

Reply:
after https://github.com/apache/spark/pull/39047, I think you can now create a literal int from the client side

Reply:
Yes, Thank for pointing out this.